src/pointer.rs

```diff
@@ -0,0 +1,427 @@
+use crate::kurbo::{Point, Size, Vec2};
+use crate::Modifiers;
+
+// NOTE: We store pen inclination as azimuth/altitude even though some platforms use the tilt x/y representation.
+// There is a small conversion cost, but azimuth/altitude is more accurate for large tilts so it's the better
+// base representation.
+#[derive(Debug, Clone, PartialEq)]
+pub struct PenInclination {
+    pub azimuth_angle: f64,
+    pub altitude_angle: f64,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct PenInclinationTilt {
+    pub tilt_x: i8,
+    pub tilt_y: i8,
+}
+
+impl PenInclination {
+    // NOTE: We store pen inclination as whatever the platform gives it to us as - either tilt x/y or azimuth_angle/altitude_angle.
+    //  It can be requested as either tilt or azimuth/angle form, and conversion is only performed on demand.
+    //  Functions are taken from:
+    //  https://www.w3.org/TR/pointerevents3/#converting-between-tiltx-tilty-and-altitudeangle-azimuthangle
+
+    pub fn from_tilt(tilt_x: i8, tilt_y: i8) -> Option<PenInclination> {
```

Comment:
I'm liking where you've landed with simplifying and canonicalizing the pen inclination stuff! 👍